harper-core/src/linting/spelled_numbers.rs

```diff
@@ -64,14 +70,18 @@ fn spell_out_number(num: u64) -> Option<String> {
         70 => "seventy".to_string(),
         80 => "eighty".to_string(),
         90 => "ninety".to_string(),
-        100 => "one hundred".to_string(),
+        hundred if hundred % 100 == 0 => {
+            format!("{} hundred", spell_out_number(hundred / 100).unwrap())
+        }
         _ => {
-            let parent = (num / 10) * 10;
-            let child = num % 10;
+            let n = 10u64.pow((num as f32).log10() as u32);
+            let parent = (num / n) * n; // truncate
+            let child = num % n;
 
             format!(
-                "{} {}",
+                "{}{}{}",
                 spell_out_number(parent).unwrap(),
+                if num <= 99 { '-' } else { ' ' },
```

Comment:
This is because numbers between twenty-one and ninety-nine should be hyphenated.

Reply:
Toss this info in a comment in the code.